libraries/AP_Scripting/applets/follow-target-send.lua

```diff
@@ -0,0 +1,119 @@
+--
+-- How To Use
+-- 1. copy this script to the autopilot's "scripts" directory
+-- 2. within the "scripts" directory create a "modules" directory
+-- 3. copy the MAVLink/mavlink_msgs_xxx files to the "scripts" directory
+-- 4. the FOLLOW_TARGET message will be published at 10hz
+
+-- load mavlink message definitions from modules/MAVLink directory
+local mavlink_msgs = require("MAVLink/mavlink_msgs")
+
+-- global definitions
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+local UPDATE_INTERVAL_MS = 100                  -- update at about 10hz
+local FOLLOW_TARGET_CAPABILITIES = {POS=2^0, VEL=2^1, ACCEL=2^2, ATT_RATES=2^3}
+
+ -- setup script specific parameters
+local PARAM_TABLE_KEY = 88
+local PARAM_TABLE_PREFIX = "FOLT_"
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 2), 'could not add param table')
+
+-- add a parameter and bind it to a variable
+local function bind_add_param(name, idx, default_value)
+    assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', PARAM_TABLE_PREFIX .. name))
+    return Parameter(PARAM_TABLE_PREFIX .. name)
+end
+
+--[[
+  // @Param: FOLT_ENABLE
+  // @DisplayName: Follow Target Send Enable
+  // @Description: Follow Target Send Enable
+  // @Values: 0:Disabled,1:Enabled
+  // @User: Standard
+--]]
+local FOLT_ENABLE = bind_add_param("ENABLE", 1, 1)
+
+--[[
+  // @Param: FOLT_MAV_CHAN
+  // @DisplayName: Follow Target Send MAVLink Channel
+  // @Description: MAVLink channel to which FOLLOW_TARGET should be sent
+  // @Range: 0 10
+  // @User: Standard
+--]]
+local FOLT_MAV_CHAN = bind_add_param("MAV_CHAN", 2, 0)
+
+-- initialize MAVLink rx with number of messages, and buffer depth
+mavlink:init(2, 5)
+
+-- send FOLLOW_TARGET message
+local function send_follow_target_msg()
+
+    -- get vehicle location
+    local curr_loc = ahrs:get_location()
+    if curr_loc == nil then
+        do return end
+    end
+
+    -- get vehicle target velocity in m/s in NED frame
+    local vel_target_NED = poscontrol:get_vel_target()
+    if vel_target_NED == nil then
+        vel_target_NED = Vector3f()
+    end
+
+    -- get vehicle target acceleration in m/s/s in NED frame
+    local accel_target_NED = poscontrol:get_accel_target()
+    if accel_target_NED == nil then
+        accel_target_NED = Vector3f()
+    end
+
+    -- get vehicle current attitude as quaternion
+    local attitude_quat = ahrs:get_quaternion()
+    if attitude_quat == nil then
+        attitude_quat = Quaternion()
+    end
+
+    -- get vehicle rotation rates
+    local curr_rot_rate = ahrs:get_gyro()
+    if curr_rot_rate == nil then
+        curr_rot_rate = Vector3f()
+    end
+    local curr_rot_rate_NED = ahrs:body_to_earth(curr_rot_rate)
+
+    -- prepare FOLLOW_TARGET message
+    local follow_target_msg = {}
+    follow_target_msg.timestamp = millis():toint() * 1000
```

Comment:
I think millis is correct, no need for the * 1000?

We do support 64bit numbers now. We could add a binding for millis64.

Reply:
thanks @IamPete1 for the review.  I thought of this but then wondered if the millis64 wrapped more quickly than millis()?  Not that most vehicles are powered up long enough to wrap but..

Reply:
oh my, you're right!  fixed!